examples/pipeline/wav2letter.py

```diff
@@ -0,0 +1,1179 @@
+import torch.multiprocessing as mp
+
+# https://github.com/pytorch/pytorch/issues/13883
+if __name__ == '__main__':
+    mp.set_start_method('forkserver')
+
+import argparse
+import collections
+import cProfile
+import hashlib
+import itertools
+import math
+import os
+import pprint
+import pstats
+import random
+import re
+import shutil
+import signal
+import statistics
+import string
+from array import array
+from collections import defaultdict
+from datetime import datetime
+from io import StringIO
+from typing import Optional
+
+import matplotlib
+import torch
+import torch.distributed as dist
+import torchaudio
+from matplotlib import pyplot as plt
+from torch import nn, topk
+from torch.optim import SGD, Adadelta, Adam
+from torch.optim.lr_scheduler import ExponentialLR, ReduceLROnPlateau
+from torch.utils.data import DataLoader
+from torchaudio.datasets import LIBRISPEECH, SPEECHCOMMANDS
+from torchaudio.datasets.utils import bg_iterator, diskcache_iterator
+from torchaudio.models.wav2letter import Wav2Letter
+from torchaudio.transforms import MFCC, Resample
+from tqdm.notebook import tqdm as tqdm
+
+from tabulate import tabulate
+
+
+print("start time: {}".format(str(datetime.now())), flush=True)
+
+try:
+    get_ipython().run_line_magic('matplotlib', 'inline')
+    in_notebook = True
+except NameError:
+    matplotlib.use("Agg")
+    in_notebook = False
+
+# Empty CUDA cache
+torch.cuda.empty_cache()
+
+# Profiling performance
+pr = cProfile.Profile()
+pr.enable()
+
+
+# Create argument parser
+parser = argparse.ArgumentParser()
```

Comment:
I'd move this to the bottom of the file and guard it by ```if __name__ ==  "__main__":```and create a "main" function that accepts args as an object. This way this code won't run on import and can be executed from another script. It's also more canonical

Reply:
Defining variables in global scope makes it very hard to refactor the code later when this script is found useful. And often times unintended variable shadowing or access to global variable occurs. So it's highly recommended to refactor the code.

`torchvision` does very good job in their reference code. https://github.com/pytorch/vision/blob/master/references/detection/train.py